lib/Auth/Source/X509userCert.php

```diff
@@ -92,6 +92,12 @@ public function authFailed(&$state): void
         $t->data['loginurl'] = Utils\HTTP::getSelfURL();
         $t->data['errorcode'] = $state['authX509.error'];
         $t->data['errorcodes'] = Error\ErrorCodes::getAllErrorCodeMessages();
+        $t->data['errortitle'] = $t->data['errorcode'];
+        if (array_key_exists($t->data['errorcode'], $t->data['errorcodes'])) {
+            $t->data['errordescr'] = $t->data['errorcodes'][$t->data['errorcode']];
+        } else {
+            $t->data['errordescr'] = $t->data['errorcode'];
+        }
```

Comment:
This is the result of refactoring towards twig templates that was incomplete. To fully clean it up I propose the following:
- Do not set `errorcode` and `errorcodes` as template variables but just as local variables and use those in the code in the added lines above.
- Change the `if` condition in the twig template to test for presence of `errordescr`.